### PR TITLE
Startup order added

### PR DESCRIPTION
**Pull request by chernikovmaksym**
Java-server branch changed from development to master
Admin-console version changed to latest stable

### CODE REVIEW

File: rdbms-image/devicehive-backend/Dockerfile [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
 
 ENV DH_VERSION="3.0.0-SNAPSHOT"
 ENV MAVEN_VERSION="3.3.3"
 
-RUN apk update && apk upgrade && \
-    apk add --no-cache bash git openssh curl
+RUN apt-get update
+RUN apt-get install -y netcat # For startup order in devicehive-start.sh
+RUN apt-get install -y wget
+RUN apt-get install -y git
```
Comment (filatovw) [inline rdbms-image/devicehive-backend/Dockerfile]:
```{.sh} 
RUN apt-get update -y \
    && apt-get install -y \
        netcat \
        wget \
        git \
    && apt-get autoremove -y \
    && apt-get clean
```
Reduce number of layers

Comment (chernikovmaksym) [inline rdbms-image/devicehive-backend/Dockerfile]: fixed

File: devicehive-admin/Dockerfile [hunk 2]
```diff
@@ -8,9 +8,9 @@ RUN mkdir -p /opt/devicehive/admin
 ADD admin-start.sh /opt/devicehive/
 
 #installing devicehive admin console
-ADD https://github.com/devicehive/devicehive-admin-console/archive/3.0.0.tar.gz /tmp
-RUN tar -C /tmp -xzf /tmp/3.0.0.tar.gz
-RUN cp -r /tmp/devicehive-admin-console-3.0.0/* /opt/devicehive/admin/
+ADD https://github.com/devicehive/devicehive-admin-console/archive/stable.tar.gz /tmp
+RUN tar -C /tmp -xzf /tmp/stable.tar.gz
+RUN cp -r /tmp/devicehive-admin-console-stable/* /opt/devicehive/admin/
```
Comment (aleksey-suprun) [inline devicehive-admin/Dockerfile]: This and previous lines *must* be merged into single RUN instruction. Also as I remember, ADD is able to extract tar.gz automatically. Please double-check that.

File: rdbms-image/devicehive-backend/Dockerfile [hunk 3]
```diff
@@ -1,12 +1,17 @@
-FROM frolvlad/alpine-oraclejdk8
+FROM picoded/ubuntu-openjdk-8-jdk
```
Comment (aleksey-suprun) [inline rdbms-image/devicehive-backend/Dockerfile]: What is the motivation of this change? Why OpenJDK? Why not Alpine linux?

File: rdbms-image/devicehive-backend/Dockerfile [hunk 4]
```diff
@@ -18,7 +23,7 @@ RUN cd /usr/share \
 ENV MAVEN_HOME /usr/share/maven
 
 # download devicehive and build .jar
-RUN git clone -b development https://github.com/devicehive/devicehive-java-server.git /opt/devicehive && \
+RUN git clone -b master https://github.com/devicehive/devicehive-java-server.git /opt/devicehive && \
```
Comment (aleksey-suprun) [inline rdbms-image/devicehive-backend/Dockerfile]: We should not clone repo and build the artifact inside Dockerfile. It's better to use prebuilt jar file. Because this cause a lot of unnecessary garbage inside container and it became too heavy.

File: rdbms-image/devicehive-frontend/Dockerfile [hunk 5]
```diff
@@ -1,12 +1,17 @@
-FROM frolvlad/alpine-oraclejdk8
+FROM picoded/ubuntu-openjdk-8-jdk
```
Comment (aleksey-suprun) [inline rdbms-image/devicehive-frontend/Dockerfile]: The same question as above

File: rdbms-image/devicehive-frontend/Dockerfile [hunk 6]
```diff
@@ -18,7 +23,7 @@ RUN cd /usr/share \
 ENV MAVEN_HOME /usr/share/maven
 
 # download devicehive and build .jar
-RUN git clone -b development https://github.com/devicehive/devicehive-java-server.git /opt/devicehive && \
+RUN git clone -b master https://github.com/devicehive/devicehive-java-server.git /opt/devicehive && \
```
Comment (aleksey-suprun) [inline rdbms-image/devicehive-frontend/Dockerfile]: See similar comments above

File: riak-image/devicehive-backend/Dockerfile [hunk 7]
```diff
@@ -1,12 +1,18 @@
-FROM frolvlad/alpine-oraclejdk8
+FROM picoded/ubuntu-openjdk-8-jdk
```
Comment (aleksey-suprun) [inline riak-image/devicehive-backend/Dockerfile]: The same question about versions

File: riak-image/devicehive-backend/Dockerfile [hunk 8]
```diff
@@ -18,7 +24,7 @@ RUN cd /usr/share \
 ENV MAVEN_HOME /usr/share/maven
 
 # download devicehive and build .jar
-RUN git clone -b development https://github.com/devicehive/devicehive-java-server.git /opt/devicehive && \
+RUN git clone -b master https://github.com/devicehive/devicehive-java-server.git /opt/devicehive && \
```
Comment (aleksey-suprun) [inline riak-image/devicehive-backend/Dockerfile]: The same as above

File: riak-image/devicehive-frontend/Dockerfile [hunk 9]
```diff
@@ -1,12 +1,17 @@
-FROM frolvlad/alpine-oraclejdk8
+FROM picoded/ubuntu-openjdk-8-jdk
```
Comment (aleksey-suprun) [inline riak-image/devicehive-frontend/Dockerfile]: The same as above

File: riak-image/devicehive-frontend/Dockerfile [hunk 10]
```diff
@@ -18,7 +23,7 @@ RUN cd /usr/share \
 ENV MAVEN_HOME /usr/share/maven
 
 # download devicehive and build .jar
-RUN git clone -b development https://github.com/devicehive/devicehive-java-server.git /opt/devicehive && \
+RUN git clone -b master https://github.com/devicehive/devicehive-java-server.git /opt/devicehive && \
```
Comment (aleksey-suprun) [inline riak-image/devicehive-frontend/Dockerfile]: The same as above

File: devicehive-admin/Dockerfile (hunk 2 above)
Comment (chernikovmaksym) [inline devicehive-admin/Dockerfile]: Lines merged into single RUN instruction.
ADD is not able to extract remote files, only local ones.

File: rdbms-image/devicehive-backend/Dockerfile (hunk 4 above)
Comment (chernikovmaksym) [inline rdbms-image/devicehive-backend/Dockerfile]: This repo is often used for developers to run quickly the latest java-server version, not only release archives. So building .jars from source code gives more flexibility in choosing branches and versions. I think it needs to be discussed with the team.

File: rdbms-image/devicehive-backend/Dockerfile (hunk 3 above)
Comment (chernikovmaksym) [inline rdbms-image/devicehive-backend/Dockerfile]: Motivation was to use netcat ubuntu options for startup order. And OpenJDK Ubuntu was the most lightweight image I found. But in the latest commit I achieved the same behavior with netcat-openbsd for linux, so for saving disk space I returned to Alpine.

File: devicehive-admin/Dockerfile [hunk 11]
```diff
@@ -11,7 +11,8 @@ ADD admin-start.sh /opt/devicehive/
 ADD https://github.com/devicehive/devicehive-admin-console/archive/stable.tar.gz /tmp
 RUN tar -C /tmp -xzf /tmp/stable.tar.gz \
     && cp -r /tmp/devicehive-admin-console-stable/* /opt/devicehive/admin/ \
```
Comment (aleksey-suprun) [inline devicehive-admin/Dockerfile]: `/tmp/devicehive-admin-console-stable/` also should be removed. This directory is just copied and as a result we have 2 identical directories in image

File: rdbms-image/devicehive-backend/Dockerfile [hunk 12]
```diff
@@ -4,27 +4,27 @@ MAINTAINER devicehive
...
-    && ln -s /usr/share/maven/bin/mvn /usr/bin/mvn
-
-ENV MAVEN_HOME /usr/share/maven
-
-RUN mkdir -p /opt/devicehive
-
-# download devicehive and build .jar
-RUN git clone -b master https://github.com/devicehive/devicehive-java-server.git /opt/devicehive \
-    && cd /opt/devicehive \
+    && apk add --no-cache bash netcat-openbsd curl \
```
Comment (aleksey-suprun) [inline rdbms-image/devicehive-backend/Dockerfile]: This line should be in separate layer (RUN)

File: rdbms-image/devicehive-backend/Dockerfile [hunk 13]
```diff
@@ -4,27 +4,27 @@ MAINTAINER devicehive
...
+    && apk add --virtual .build-dependencies ca-certificates git openssh openssl make \
```
Comment (aleksey-suprun) [inline rdbms-image/devicehive-backend/Dockerfile]: Why we need `make` here?

File: rdbms-image/devicehive-backend/Dockerfile [hunk 14]
```diff
@@ -4,27 +4,27 @@ MAINTAINER devicehive
 
 ENV DH_VERSION="3.0.0-SNAPSHOT"
 ENV MAVEN_VERSION="3.3.3"
+ENV MAVEN_HOME /usr/share/maven/
```
Comment (aleksey-suprun) [inline rdbms-image/devicehive-backend/Dockerfile]: Usually maven home env var names `M2_HOME `

File: rdbms-image/devicehive-backend/Dockerfile [hunk 15]
```diff
@@ -4,27 +4,27 @@ MAINTAINER devicehive
...
+    && ln -s /usr/share/maven/bin/mvn /usr/bin/mvn \
+    && mkdir -p /opt/devicehive \
+    && git clone -b master https://github.com/devicehive/devicehive-java-server.git /opt/devicehive/source \
+    && cd /opt/devicehive/source \
     && mvn clean package -Pbooted-rdbms,shim-kafka,!booted-riak -DskipTests \
-    && mv /opt/devicehive/devicehive-backend/target/devicehive-backend-${DH_VERSION}-boot.jar /opt/devicehive/
+    && mv /opt/devicehive/source/devicehive-backend/target/devicehive-backend-${DH_VERSION}-boot.jar /opt/devicehive/ \
+    && rm -rf /opt/devicehive/source \
+    && rm -rf /usr/share/maven/ \
+    && apk del .build-dependencies
```
Comment (aleksey-suprun) [inline rdbms-image/devicehive-backend/Dockerfile]: Should we also delete `~/.m2` folder with all cached jars?
